src/TorchSharp/Tensor/Tensor.cs

```diff
@@ -4122,6 +4136,74 @@ public Tensor std(long[] dimensions, bool unbiased = true, bool keepDimension =
                 }
             }
 
+            /// <summary>Calculates the standard deviation of all elements in the tensor.</summary>
+            /// <remarks>
+            /// If <paramref name="unbiased" /> is <value>true</value>, Bessel’s correction will be used.
+            /// Otherwise, the sample deviation is calculated, without any correction.
+            /// </remarks>
+            /// <param name="dimension">The dimension to reduce.</param>
+            /// <param name="unbiased">Whether to use Bessel’s correction (δN=1).</param>
+            /// <param name="keepDimension">Whether the <see cref="Tensor">output tensor</see> has <paramref name="dimension" /> retained or not.</param>
+            /// <param name="type"></param>
+            /// <returns>The <see cref="Tensor">output tensor</see>.</returns>
+            [System.Diagnostics.Contracts.Pure]
+            public Tensor std(long dimension, bool unbiased = true, bool keepDimension = false, ScalarType? type = null)
+                => std(new[] { dimension }, unbiased, keepDimension, type);
+
+            [DllImport("LibTorchSharp")]
+            static extern IntPtr THSTensor_std_mean(IntPtr tensor);
+
+            /// <summary>
+            /// Calculates the standard deviation and mean of all elements in the tensor.
+            /// </summary>
+            [System.Diagnostics.Contracts.Pure]
+            public Tensor std_mean()
+            {
+                var res = THSTensor_std_mean(Handle);
+                if (res == IntPtr.Zero)
+                    torch.CheckForErrors();
+                return new Tensor(res);
+            }
+            
+            [DllImport("LibTorchSharp")]
+            static extern IntPtr THSTensor_std_mean_along_dimensions(IntPtr tensor, IntPtr dimensions, int length, bool unbiased, bool keepdim);
+
+            /// <summary>Calculates the standard deviation and mean of all elements in the tensor.</summary>
+            /// <remarks>
+            /// If <paramref name="unbiased" /> is <value>true</value>, Bessel’s correction will be used.
+            /// Otherwise, the sample deviation is calculated, without any correction.
+            /// </remarks>
+            /// <param name="dimensions">The dimensions to reduce.</param>
+            /// <param name="unbiased">Whether to use Bessel’s correction (δN=1).</param>
+            /// <param name="keepDimension">Whether the <see cref="Tensor">output tensor</see> has dim retained or not.</param>
+            /// <param name="type"></param>
+            /// <returns>The <see cref="Tensor">output tensor</see>.</returns>
+            [System.Diagnostics.Contracts.Pure]
+            public Tensor std_mean(long[] dimensions, bool unbiased = true, bool keepDimension = false, ScalarType? type = null)
```

Comment:
I've started adding overloads with `ReadOnlySpan<long>` in addition to the `long[]` version. That helps make the other overloads marginally faster, since they can use 'stackalloc' instead of 'new' when creating the array to pass in for the dimensions argument.

The easiest thing to do is to duplicate the body with the unsafe code -- having the `long[]` version call the `ReadOnlySpan<long>` version risks infinite recursion (which I discovered recently).

Reply:
seems like only the `ReadOnlySpan<long>` version is required, since an `Array<T>` can be converted to an `ReadOnlySpan<T>` implicitely. I've removed the array version.

Reply:
> seems like only the `ReadOnlySpan<long>` version is required, since an `Array<T>` can be converted to an `ReadOnlySpan<T>` implicitely. I've removed the array version.

Unfortunately, not in F#. The array version is required in order for F# array literals to work.